atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,140 @@
+        """
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        labels = Labels(dockerfile.labels)
+        try:
+            _, operator_label = labels.get_name_and_value(Labels.LABEL_TYPE_OPERATOR_MANIFESTS)
+        except KeyError:
+            operator_label = 'false'
+        return operator_label.lower() == 'true'
+
+    def is_orchestrator(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        if self.operator_manifests_extract_platform != self.platform:
+            self.log.info("Only platform [%s] will upload operators metadata. Skipping",
+                          self.operator_manifests_extract_platform)
+            return False
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+
+        This plugin extracts the operator manifest files from an image, saves
+        them as a zip archive, and returns its path
+
+        :return: str, path to operator manifests zip file
+        """
+        if not self.should_run():
+            return
+
+        manifests_archive_dir = tempfile.mkdtemp()
+        image = self.workflow.image
+        # As in flatpak_create_oci, we specify command to prevent possible docker daemon errors.
+        container_dict = self.tasker.d.create_container(image, command=['/bin/bash'])
+        container_id = container_dict['Id']
+        try:
+            bits, stat = self.tasker.d.get_archive(container_id,
+                                                   os.path.join('/', MANIFESTS_DIR_NAME))
+        except APIError as ex:
+            self.log.error('Could not extract operator manifest files: %s', ex)
+            raise RuntimeError('Could not extract operator manifest files: %s' % ex)
+        finally:
+            self.tasker.d.remove_container(container_id)
+
+        fd, extracted_file = tempfile.mkstemp()
+        os.close(fd)
+        with open(extracted_file, 'wb') as f:
+            for chunk in bits:
+                f.write(chunk)
+
+        tar_archive = tarfile.TarFile(extracted_file)
+        tar_archive.extractall(manifests_archive_dir)
+        manifests_path = os.path.join(manifests_archive_dir, MANIFESTS_DIR_NAME)
+        manifest_files = os.listdir(manifests_path)
```

Comment:
This supports only 1 level but there is some ongoing work that manifests may have nested directories. I'd like to just archive the whole dir.

Reply:
Fixed. Also added a test case to cover it.